src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs

```diff
@@ -133,5 +133,34 @@ internal bool HasClientCredentials
         /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
         /// </summary>
         public string? UserAssignedManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow app-only tokens?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration you'll need to opt-in for the idtyp claim.
+        /// </remarks>
+        public bool ApiAllowsAppOnlyTokens { get; set; } = true;
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow guest accounts?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration for your web API, you'll need to opt-in for the acct claim.
+        /// </remarks>
+        public bool ApiAllowsGuestAccounts { get; set; } = true;
+
+        /// <summary>
+        /// Applies to web apps and web APIs. Enables to provide the list
+        /// of tenant Ids the user of whom are allowed to call the web API
+        /// or log-in to the web app.
+        /// </summary>
+        public IEnumerable<string>? AllowedTenantIds { get; set; }
```

Comment:
It is either single or MT app. Consider including the tenant ID in policy definition instead of offering a method.

Reply:
you mean an ASP.NET Core policy, @yordan-msft ?